File: 1_Easy/mathproblemgen/mathproblemgen.py
# Math equation generator
# Difficulty:
# Generate math equations and ask the user to solve the equation

# Steps:
# 1. Create 2 variables and save a number in each variable
# 2. output a string equation using the variables as the numbers

# 3. ask the user to solve the equation
# 4. check to see if the user is correct
# 5. NOTE: make sure to cast the data to match properly

# 6. replace the numbers with random numbers

# 7. repeat with subtraction, multiplication

# Bonus:
# 1. add a loop to repeat 10 times
# 2. set random number to select the type of equation seperately

import random

score = 0
problemnum = 1

for i in range(10):
    number1 = random.randint(1, 30)
    number2 = random.randint(1, 30)

    operator = random.randint(1, 4)
    #addition
    if operator == 1:
        question = str(problemnum) + ". " + str(number1) + "+" + str(number2) + "="
        answer = input(question)
        answer = int(answer)
        if answer == number1 + number2:
           print("correct")
           score = score + 1
        else:
            print("incorrect! Answer was", number1 + number2)

    #subtraction
    elif operator == 2:
        question = str(problemnum) + ". " + str(number1) + "-" + str(number2) + "="
        answer = input(question)
        answer = int(answer)
        if answer == number1 - number2:
            print("correct")
            score = score + 1
        else:
            print("incorrect! Answer was", number1 - number2)

    #multiplication
    elif operator == 3:
        question = str(problemnum) + ". " + str(number1) + "*" + str(number2) + "="
        answer = input(question)
        answer = int(answer)
        if answer == number1 * number2:
            print("correct")
            score = score + 1
        else:
            print("incorrect! Answer was", number1 * number2)

    #division
    elif operator == 4:
        question = str(problemnum) + ". " + str(number1) + "/" + str(number2) + "="
        answer = input(question)
        answer = float(answer)
        if answer == round(number1 / number2, 2):
            print("correct")
            score = score + 1
        else:
            print("incorrect! Answer was", round(number1 / number2, 2))

    #problem number
    problemnum = problemnum + 1

print("Score:", score)

File: webstuff/mathproblemgen/mathproblemgen.js
// math problem generator
// generate a random equation and ask the user to solve it

// Steps
// - generate 2 random numbers and store them in variables
// - output the eqation with the 2 numbers and a "+" in between

// - ask the user to solve the eqation with prompt
// - check to see if the user got it correct
// - NOTE: make sure the data types are the same before comparison

// - repeat for subtraction, multiplication

// - repeat for division
// - NOTE: data type needs to be a float
// - round to the closest 2 decimal points by doing some math

// Bonus
// - add randomness to randomly choose an operator for the equation
// - add a loop and keep track of the total correctly answered equation

const prompt = require("prompt-sync")();

let operator = parseInt(Math.random() * 4 + 1);
let num1 = parseInt(Math.random() * 100 + 1);
let num2 = parseInt(Math.random() * 100 + 1);

if (operator === 1) {
  console.log("Addition problem!");
  let answer = prompt(num1 + "+" + num2 + "= ");
  answer = parseInt(answer);
  if (answer === num1 + num2) {
    console.log("You got it right!");
  } else {
    console.log("Wrong, it was", num1 + num2);
  }
} else if (operator === 2) {
  console.log("Subtraction problem!");
  let answer = prompt(num1 + "-" + num2 + "= ");
  answer = parseInt(answer);
  if (answer === num1 - num2) {
    console.log("You got it right!");
  } else {
    console.log("Wrong, it was", num1 - num2);
  }
} else if (operator === 3) {
  console.log("Multiplication problem!");
  num1 = parseInt(Math.random() * 10 + 1);
  num2 = parseInt(Math.random() * 10 + 1);
  let answer = prompt(num1 + "*" + num2 + "= ");
  answer = parseInt(answer);
  if (answer === num1 * num2) {
    console.log("You got it right!");
  } else {
    console.log("Wrong, it was", num1 * num2);
  }
} else if (operator === 4) {
  console.log("Division problem!");
  num1 = parseInt(Math.random() * 10 + 1);
  num2 = parseInt(Math.random() * 10 + 1);
  let answer = prompt(num1 + "/" + num2 + "= ");
  answer = parseFloat(answer);

  answer = answer * 100;
  answer = Math.round(answer);
  answer = answer / 100;

  if (answer === num1 / num2) {
    console.log("You got it right!");
  } else {
    console.log("Wrong, it was", num1 / num2);
  }
}
